### DISCUSSION TITLE
how to export multiple different image

### DISCUSSION THREAD

**Discussion by qinghon** [go-debos/debos#512] [Q&A] [answered]
I'm trying to use debos to build an image similar to armbian, and need to image many different devices.
Most of these images only differ in devicetree/uboot

The current debos example only uses one yaml file to implement a single image export, and the action execution results are superimposed.

Is there a way for debos to make multiple slightly different images and export them?

**Comment by obbardc** [comment, marked as answer]
I'd suggest to create two recipes; one which generates a generic tarball (with the pack action); then a separate recipe which generates the hardware-specific image.

**Reply by qinghon** [reply]
YES! , This can solve a lot of problems
According to this idea, the steps should be

step1：
>  debootstrap
>  any action
>  pack

step2:
> unpack
> any different action
> deploy

But it may not be very flexible, for example:
- Only two layers are supported. If a batch of hardware is not very different and is divided into multiple groups, the same group will have many repeated actions.
- Redundant pack/unpack steps

I think the ideal process in this scenario is:
>  debootstrap
>  any action
>  commit0
> fork commit0
> another action
> commit1
> fork commit1
> another action
> export image0
> fork commit0
> another action
> export image1

Enter multiple yaml configurations from the command line and generate an execution tree, and finally export all leaf nodes.

Maybe I should fork the repository and write a demo

BTW, I am making an image for a box based on amlogic CPU. Amlogic cpu types are not too many, but there are many boxes derived from it, most of which have duplicate configurations.
